### PR TITLE
Simplify blocking-AM management in comm=ofi.

### PR DESCRIPTION
**Pull request by gbtitus**
The part of the ofi comm layer's common Active Message support that
dealt with whether or not we block waiting for the AM to finish was
unnecessarily complicated, especially for AM-mediated atomic ops where
we might want to delay that blocking until our memory consistency model
requires the atomic op to be done.  Here, refactor so that the callers
simply tell the common code whether or not to block immediately, and
handle the delayed-blocking case entirely in the AM-mediated AMO code.

This resolves https://github.com/Cray/chapel-private/issues/2817.

### CODE REVIEW

**Comment by gbtitus** [comment]
@ronawho FYI.

File: runtime/src/comm/ofi/comm-ofi.c [hunk 1]
```diff
@@ -3610,12 +3610,10 @@ typedef enum {
   am_opShutdown,                           // signal main process for shutdown
 } amOp_t;
 
-#ifdef CHPL_COMM_DEBUG
 static inline
 chpl_bool op_uses_on_bundle(amOp_t op) {
   return op == am_opExecOn || op == am_opExecOnLrg;
 }
-#endif
```
Comment (ronawho) [inline runtime/src/comm/ofi/comm-ofi.c:3618]: Why is this now required for non-debug builds too? (I see that call that uses it, but I'm wondering why the logic changes if this was mostly a non-functional code cleanup)

Comment (ronawho) [reply runtime/src/comm/ofi/comm-ofi.c:3618]: Ah, never mind -- I think it's just that we've lost context that was available at the call site inside the callee, so we need to use this to query now.

**Review by ronawho — APPROVED** [review]

File: runtime/src/comm/ofi/comm-ofi.c (hunk 1 above)
Comment (gbtitus) [reply runtime/src/comm/ofi/comm-ofi.c:3618]: Right, the new call to this is in the common code where the operator could be anything, and as it happens is the first call to it outside of debug context.

**Review by jhh67 — APPROVED** [review]

**Pull request merged**